tests/system_test.sh

```diff
@@ -71,16 +47,13 @@ for i in `seq 1 $CARBAPI_RETRIES`; do
     fi
 done
 
-if [ $i -eq $CARBAPI_RETRIES ]; then
+if [ "$i" -eq $CARBAPI_RETRIES ]; then
```

Comment:
`-eq` is for numeric comparison, hence quoting "$i" looks dodgy, since all it could achieve compared to having no quotes is that it keeps whitespaces, which is not something numeric comparison would like

Reply:
This is driven by the shellcheck warning.

Reply:
More specifically, as you say, it prevents splitting. This should not happen, of course, but this is simply defensive programming. If it does happen, it's better with quotes than without. I think this is why the linter insists it is a warning.